lib/api/bucketDeletePolicy.js

```diff
@@ -0,0 +1,51 @@
+const { errors } = require('arsenal');
+
+const metadata = require('../metadata/wrapper');
+const { metadataValidateBucket } = require('../metadata/metadataUtils');
+const collectCorsHeaders = require('../utilities/collectCorsHeaders');
+
+/**
+ * bucketDeletePolicy - Delete the bucket policy
+ * @param {AuthInfo} authInfo - Instance of AuthInfo class with requester's info
+ * @param {object} request - http request object
+ * @param {object} log - Werelogs logger
+ * @param {function} callback - callback to server
+ * @return {undefined}
+ */
+function bucketDeletePolicy(authInfo, request, log, callback) {
+    log.debug('processing request', { method: 'bucketDeletePolicy' });
+    const { bucketName, headers, method } = request;
+    const metadataValParams = {
+        authInfo,
+        bucketName,
+        requestType: 'bucketOwnerAction',
+    };
+    return metadataValidateBucket(metadataValParams, log, (err, bucket) => {
+        const corsHeaders = collectCorsHeaders(headers.origin, method, bucket);
+        if (err) {
+            log.debug('error processing request', {
+                error: err,
+                method: 'bucketDeletePolicy',
+            });
+            return callback(err, corsHeaders);
+        }
+        if (!bucket.getBucketPolicy()) {
+            log.trace('no existing bucket policy', {
+                method: 'bucketDeletePolicy',
+            });
+            // TODO: implement Utapi metric support
+            return callback(null, corsHeaders);
+        }
+        log.trace('deleting bucket policy in metadata');
+        bucket.setBucketPolicy(null);
```

Comment:
Thinking out loud here: Maybe we can introduce a new method `deleteBucketPolicy` to `BucketInfo`. I know we've already been passing in `null` to the setter in other APIs so I could be persuaded on the merits of consistency, but it seems more intuitive than passing `null` to the setter to have the effect of deletion.

Reply:
yeah, it's not a bad idea, i do think it's just keeping the consistency with our other apis and also the convention of just having getters/setters on classes 🤷‍♀